metadata/user_feedsMetaData.php

```diff
@@ -46,7 +46,7 @@
     
        array('name' =>'user_id', 'type' =>'varchar', 'len'=>'36', )
       , array('name' =>'feed_id', 'type' =>'varchar', 'len'=>'36', )
-      , array('name' =>'rank', 'type' =>'int', 'required' => false)
+      , array('name' =>'suite_rank', 'type' =>'int', 'required' => false)
```

Comment:
While this change did go into ```7.8.x```, I'm thinking it probably shouldn't have for backwards compatibility reasons (Assuming we can create a workaround). I haven't tried this out but would adding an alias here work?

*Not super necessary but it would also be nice to clean up some of this file while we're here. Attached a .txt to show what I mean (Can't wait for GitHub multi-line code suggestions).

[user_feedsMetaData.txt](https://github.com/salesagility/SuiteCRM/files/3704618/user_feedsMetaData.txt)

Reply:
I don't think alias will solve the bug. Let's think we define the field like you wrote in .txt:

```php
        [
            'name' => 'rank',
            'type' => 'int',
            'alias' => 'suite_rank',
            'required' => false
        ],
```

This will be [translated](https://github.com/salesagility/SuiteCRM/blob/master/data/SugarBean.php#L3656) into SQL:

```SQL
SELECT rank as suite_rank
```

But the problem is the name `rank` when **creating** the table:

```sql
Error creating table: users_feeds Query Failed: CREATE TABLE users_feeds (user_id varchar(36)  NULL ,feed_id varchar(36)  NULL ,rank int  NULL ,date_modified datetime  NULL ,deleted bool  DEFAULT '0' NULL  , KEY idx_ud_user_id (user_id, feed_id)) CHARACTER SET utf8 COLLATE utf8_general_ci: MySQL error 1064: You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'rank int  NULL ,date_modified datetime  NULL ,deleted bool  DEFAULT '0' NULL  , ' at line 1
```

I think we face these situations:

**If we do not fix in 7.10:**

1. SuiteCRM 7.10 installation with MySQL 8.0.x: The installation will fail.
2. SuiteCRM 7.10 installed with MySQL 5.7 and upgraded to MySQL 8.0.x: I think the upgrade process of MySQL will fail or at least will show a warning.

**If we fix in 7.10:**

1. SuiteCRM 7.10 installation with MySQL 8.0.x: The installation will work ok.

2. SuiteCRM 7.10.x (pre-fix) installed with MySQL 5.7.x and upgraded to 7.10.x (post-fix): The QR&R will perform an ```ALTER TABLE users_feeds...``` to change the name of the field `rank` to `suite_rank`

3. SuiteCRM 7.10 (post-fix) installed with MySQL 5.x and upgraded to MySQL 8.0: Will work ok.


I agree with you, we can clean up this file.